test/backend/routes.spec.js

```diff
@@ -198,58 +216,93 @@ describe('Routes - ', () => {
         .catch(done);
     });
 
-    it('index page - returns 200', function(done) {
-        GET('')
-        .then(res => {
-            assert.equal(res.status, 200);
-            done();
-        })
+    it('allows access to login page without logging in', function(done) {
+        GET('login')
+        .then(() => done())
         .catch(done);
     });
 
-    // Settings
-    it('settings/urls - returns 200 and the urls', function(done) {
-        GET('settings/urls')
-        .then(res => res.json().then(json => {
-            assert.equal(res.status, 200);
-            assert.equal(json.http, 'http://localhost:9494');
-            assert.equal(json.https, '');
-            done();
-        }))
+    it('allows access to connections page without logging in', function(done) {
+        GET('')
+        .then(() => done())
         .catch(done);
     });
 
-    it('GET /settings returns some of the settings', function(done) {
-        saveSetting(
-            'USERS',
-            [{'username': 'chris', 'accessToken': 'lahlahlemons'}]
-        );
+    it('does not allow access to settings when not logged in', function(done) {
         GET('settings')
         .then(res => res.json().then(json => {
+            assert.equal(res.status, 401);
             assert.deepEqual(json, {
-                'PLOTLY_URL': 'https://plot.ly',
-                'USERS': ['chris']
+                'code': 'InvalidCredentials',
+                'message': 'Please login to access this page.'
             });
             done();
         })).catch(done);
     });
 
-    it('PATCH /settings sets some settings', function(done) {
-        const newSettings = {
-            'PLOTLY_API_DOMAIN': 'acme.plot.ly',
-            'PLOTLY_API_SSL_ENABLED': false
+    it('allows access to settings when logged in', function(done) {
+
+        setCookies().then(res => {
+
+            GET('settings')
+            .then(res => {
+                assert.equal(res.status, 200);
+                done();
+            }).catch(done);
+        });
+
+    });
```

Comment:
could we add some tests that test the behaviour of:
- making a request after the 5 min expiry token expires makes another request to plotly
- making a request within the 5 min expiry doesn't make a request to plotly
- if a user is revoked within 5 minutes then they can still access the connector but after the token expires they are denied access
- loop through all of the `GET` endpoints and verify that 401s are returned. i believe that you can access the list of routes dynamically with `server.router.mounts`
- requests to `/`, `/login/`, `/static/`, and `/oauth2/` return in 200s